pandas/_libs/tslibs/parsing.pyx

```diff
@@ -824,17 +824,11 @@ def format_is_iso(f: str) -> bint:
     Generally of form YYYY-MM-DDTHH:MM:SS - date separator can be different
     but must be consistent.  Leading 0s in dates and times are optional.
     """
-    iso_template = "%Y{date_sep}%m{date_sep}%d{time_sep}%H:%M:%S{micro_or_tz}".format
+    iso_regex = \
+        r"^\d{4}(-\d{2}(-\d{2}(T\d{2}:\d{2}:\d{2}(\.\d+)?(([+-]\d{2}:\d{2})|Z)?)?)?)?$"
```

Comment:
thanks for working on this - to keep it readable, how about making this a verbose  with comments?

see https://stackoverflow.com/a/13852096/4451315

Reply:
@MarcoGorelli added some comments for explaining the **regular expression** and how it's evaluating